includes/main.inc.php

```diff
@@ -156,14 +156,17 @@ function stripslashes_deep($value) {
 	}
 }
 
-if (isset($_GET['toggle_view'])) {
-	if (isset($_SESSION[$settings['session_prefix'].'usersettings']) && $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] == 0) {
-		$_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = 1;
-		setcookie($settings['session_prefix'].'usersettings', '1.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'], TIMESTAMP + (3600 * 24 *$settings['cookie_validity_days']));
-	} elseif (isset($_SESSION[$settings['session_prefix'].'usersettings']) && $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] == 1) {
-    $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = 0;
-    setcookie($settings['session_prefix'].'usersettings','0.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'],TIMESTAMP+(3600*24*$settings['cookie_validity_days']));
+if (isset($_GET['toggle_view']) or isset($_GET['main_view'])) {
+	if (isset($_GET['toggle_view'])) {
+		if (isset($_SESSION[$settings['session_prefix'].'usersettings']) && $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] == 0) {
+			$_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = 1;
+		} else {
+			$_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = 0;
+		}
+	} else if (isset($_GET['main_view']) and in_array($_GET['main_view'], array(0, 1))) {
+		$_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = intval($_GET['main_view']);
 	}
+	setcookie($settings['session_prefix'].'usersettings', $_SESSION[$settings['session_prefix'].'usersettings']['user_view'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'],TIMESTAMP+(3600*24*$settings['cookie_validity_days']));
```

Comment:
The `setcookie()` should be moved to the if-else because `$_SESSION[$settings['session_prefix'].'usersettings']` may in some cases not exists.

Reply:
IMHO the cookie-generation should stay outsife the if-else-block to prevent doubled or multipled notation. But you are right, when the existence of `$_SESSION[$settings['session_prefix'].'usersettings']` is not ensured, we should aks for it in a separate if-clause.

Reply:
I think I solved the issue (see b4c34fcb07). The script now makes the call of `setkookie` dependent from the existence of `$_SESSION[$settings['session_prefix'].'usersettings']` but it keeps the call independent from the previous `if`-`if else`-block.